aurora-locker/src/AuroraSdk.sol

```diff
@@ -0,0 +1,177 @@
+import "./Utils.sol";
+import "openzeppelin-contracts/token/ERC20/IERC20.sol";
+
+// Address of Cross Contract Call precompile in Aurora.
+// It allows scheduling new promises to NEAR contracts.
+address constant XCC_PRECOMPILE = 0x516Cded1D16af10CAd47D6D49128E2eB7d27b372;
+// Address of predecessor account id precompile in Aurora.
+// It allows getting the predecessor account id of the current call.
+address constant PREDECESSOR_ACCOUNT_ID_PRECOMPILE = 0x723FfBAbA940e75E7BF5F6d61dCbf8d9a4De0fD7;
+// Address of predecessor account id precompile in Aurora.
+// It allows getting the current account id of the current call.
+address constant CURRENT_ACCOUNT_ID_PRECOMPILE = 0xfeFAe79E4180Eb0284F261205E3F8CEA737afF56;
+// Addresss of promise result precompile in Aurora.
+address constant PROMISE_RESULT_PRECOMPILE = 0x0A3540F79BE10EF14890e87c1A0040A68Cc6AF71;
+// Address of wNEAR ERC20 on mainnet
+address constant wNEAR_MAINNET = 0x4861825E75ab14553E5aF711EbbE6873d369d146;
+
+struct NEAR {
+    /// Wether the represenative NEAR account id for this contract
+    /// has already been created or not. This is required since the
+    /// first cross contract call requires attaching extra deposit
+    /// to cover storage staking balance.
+    bool initialized;
+    /// Address of wNEAR token contract. It is used to charge the user
+    /// required tokens for paying NEAR storage fees and attached balance
+    /// for cross contract calls.
+    ///
+    /// On mainnet wNEAR address is: 0x4861825E75ab14553E5aF711EbbE6873d369d146
+    IERC20 wNEAR;
+}
+
+library AuroraSdk {
+    using Codec for bytes;
+    using Codec for PromiseCreateArgs;
+    using Codec for PromiseWithCallback;
+    using Codec for Borsh.Data;
+    using Borsh for Borsh.Data;
+
+    /// Default configuration for mainnet.
+    function mainnet() public pure returns (NEAR memory) {
+        return NEAR(false, IERC20(wNEAR_MAINNET));
+    }
+
+    /// Compute NEAR represtentative account for the given Aurora address.
+    /// This is the NEAR account created by the cross contract call precompile.
+    function nearRepresentative(address account) public returns (string memory) {
+        return string(abi.encodePacked(Utils.bytesToHex(abi.encodePacked((bytes20(account)))), ".", currentAccountId()));
+    }
+
+    /// Compute implicity Aurora Address for the given NEAR account.
+    function implicitAuroraAddress(string memory accountId) public pure returns (address) {
+        // TODO: Verify this returns the last 20 bytes
+        return address(bytes20(keccak256(bytes(accountId))));
+    }
+
+    /// Compute the implicit Aurora address of the represenative NEAR account
+    /// for the given Aurora address. Useful when a contract wants to call
+    /// itself via a callback using cross contract call precompile.
+    function nearRepresentitiveImplicitAddress(address account) public returns (address) {
+        return implicitAuroraAddress(nearRepresentative(account));
+    }
```

Comment:
We talked about this briefly before, how conceptually this function feels like it _should_ be the identity function (`return account`). This implementation is correct as is now, but I wonder if we shouldn't still implement a special case in the `implicitAuroraAddress` function (and the corresponding function in the engine as well) where accounts of the form `{address}.{current_account}` are mapped to `{address}` instead of being hashed.

Reply:
Let's bring this up with the engine team. I think it makes sense, but at the same time, it is more complexity added in case later we decide to deploy multiple instances of the engine for example.